### PR TITLE
fix(replace): support running against fork

### PR DESCRIPTION
**Pull request by bcoe**
When PRs are made from a fork, [such as this](https://github.com/googleapis/repo-automation-bots/pull/2349), the replacement logic does not work correctly because the branch exists on the upstream repository, not the local repository.

This PR switches to using the `pull_request` payload for replacement logic, which should work.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -49,11 +50,26 @@ async function main () {
...
-      pattern: new RegExp(`(github.com/${GITHUB_REPOSITORY}/.*/)(${GITHUB_BASE_REF})/(.*)`),
-      replacement: `$1${GITHUB_HEAD_REF}/$3`
-    });
+    const { GITHUB_HEAD_REF, GITHUB_BASE_REF, GITHUB_REPOSITORY, GITHUB_EVENT_PATH } = process.env;
+    // Read pull_request payload and use it to determine head user/repo:
+    if (GITHUB_EVENT_PATH) {
+      try {
+        const payloadRaw = await readFile(GITHUB_EVENT_PATH, 'utf8');
+        const payload = JSON.parse(payloadRaw);
+        if (payload?.pull_request?.head) {
```
Comment (JustinBeckwith) [inline src/index.js:59]: This is annoying, but could I think GitHub Actions built this way are stuck on node.js 12, where optional chaining won't work.  I should probably use Node 12 in my CI.

**Comment by JustinBeckwith** [comment]
Also annoying - right now my automation (ironically) seems to only work when I push to a PR.  Could I trouble you to run `npm run build`, and commit the changes to this PR as well?

**Comment by bcoe** [comment]
@JustinBeckwith edits made.

**Review by JustinBeckwith — APPROVED** [review]

File: src/index.js [hunk 2]
```diff
@@ -49,11 +50,26 @@ async function main () {
...
+        const payload = JSON.parse(payloadRaw);
+        if (payload && payload.pull_request && payload.pull_request.head) {
+          const repo = payload.pull_request.head.repo.full_name;
+          core.info(`rewrite repo to ${repo}`);
+          config.urlRewriteExpressions.push({
+            pattern: new RegExp(`github.com/${GITHUB_REPOSITORY}(/.*/)(${GITHUB_BASE_REF})/(.*)`),
+            replacement: `github.com/${repo}$1${GITHUB_HEAD_REF}/$3`
+          });
+        } else {
+          core.warn('unexpected payload structure', payload);
```
Comment (XhmikosR) [inline src/index.js:67]: Does it make sense to output the warning when the action runs on a  branch push, though? I know it's just a warning, but now it's emitted for all pushes to branches which don't have a PR.

**Pull request merged**